src/Hng.Domain/Common/Result.cs

```diff
@@ -0,0 +1,20 @@
+namespace Hng.Domain.Common;
+
+public class Result<T>
```

Comment:
The `Result<T>` class is more about handling the mechanics of application flow so it belongs in the `Application` layer. Could you consider moving it to ensure the Domain layer remains focused on entities and their corresponding configurations/core business logic?